src/main/kotlin/dk/cachet/carp/webservices/study/service/CoreRecruitmentService.kt

```diff
@@ -53,16 +55,19 @@ class CoreRecruitmentService
         for (participantGroupStatus in participantGroupStatusList) {
 
             val participantAccounts = arrayListOf<ParticipantAccount>()
+            val deploymentStatus = coreDeploymentService.instance.getStudyDeploymentStatus(participantGroupStatus.id)
             participantGroupStatus.participants.map {
                 val participantAccount = ParticipantAccount.fromParticipant(it)
                 val account = accountService.findByAccountIdentity(it.accountIdentity)
                 if (account != null) {
+                    val lastDataUpload = idataStreamService.getLatestUpdatedAt(participantGroupStatus.id)
                     participantAccount.lateInitFrom(account)
+                    participantAccount.dateOfLastDataUpload = lastDataUpload
                 }
                 participantAccounts.add(participantAccount)
             }
 
-            val info = ParticipantGroupInfo(participantGroupStatus.id, participantAccounts)
+            val info = ParticipantGroupInfo(participantGroupStatus.id, deploymentStatus, participantAccounts)
```

Comment:
Yes, we should defo sit down and think about it. I think this works tho. do you suppose we should halt merging before there's a test for this? (this might need prob a new branch, as I looked at the status of tests in general)

Reply:
No I don't think so, the others on the frontend kind of rely on this, I just noted this down for future development.